src/app/InteractionModelEngine.cpp

```diff
@@ -102,6 +95,26 @@ void InteractionModelEngine::Shutdown()
         }
     }
 
+    //
+    // We hold weak references to ReadClient objects. The application ultimately
+    // actually owns them, so it's on them to eventually shut them down and free them
+    // up.
+    //
+    // However, we should null out their pointers back to us at the very least so that
+    // at destruction time, they won't attempt to reach back here to remove themselves
+    // from this list.
+    //
+    for (auto * readClient = mpActiveReadClientList; readClient != nullptr; readClient = readClient->GetNextClient())
+    {
+        readClient->mpImEngine = nullptr;
+        readClient->SetNextClient(nullptr);
```

Comment:
This is not going to work right, because the next thing we will try to do is `readClient = readClient->GetNextClient()` and that will return null.

We should probably grab the `next` pointer before the `SetNextClient` call, then assign it to `readClient` after the call, and have the step part of the for loop empty?